plugin-core/src/main/java/com/salesforce/bazel/eclipse/builder/BazelErrorStreamObserver.java

```diff
@@ -0,0 +1,143 @@
+    private final IProject rootProject;
+    private static final BazelOutputParser outputParser = new BazelOutputParser();
+    private final Queue<String> errorLines = new LinkedList<>();
+    private static final LogHelper LOG = LogHelper.log(BazelErrorStreamObserver.class);
+    static final String UNKNOWN_PROJECT_ERROR_MSG_PREFIX = "ERROR IN UNKNOWN PROJECT: ";
+
+    public BazelErrorStreamObserver(final IProgressMonitor monitor, final Map<BazelLabel, IProject> labelToProject,
+            IProject rootProject) {
+        this.monitor = monitor;
+        this.labelToProject = labelToProject;
+        this.rootProject = rootProject;
+    }
+    
+    /**
+     * Starts the observer, clears Problems View for every project
+     */
+    @Override
+    public void startObserver() {
+        final Set<IProject> projectSet = new HashSet<>(this.labelToProject.values());
+        for (IProject project : projectSet) {
+            BazelMarkerSupport.publishToProblemsView(project, Collections.emptyList(), monitor);
+        }
+    }
+
+    /**
+     * Launch a thread to parse latest error line and publish to Problems View if applicable
+     */
+    @Override
+    public void update(String error) {
+        synchronized (this.errorLines) {
+            this.errorLines.add(error);
+        }
+        new Thread(() -> updateProblemsView()).start();
+    }
+
+    private void updateProblemsView() {
+        String latestLine = null;
+        synchronized (this.errorLines) {
+            latestLine = this.errorLines.remove();
+        }
+        
+        List<BazelBuildError> bazelBuildErrors = outputParser.getErrorBazelMarkerDetails(latestLine);
+        if (! bazelBuildErrors.isEmpty()) {
+            Multimap<IProject, BazelBuildError> projectToErrors =
+                    assignErrorsToOwningProject(bazelBuildErrors, this.labelToProject, this.rootProject);
+            for (IProject project : projectToErrors.keySet()) {
+                BazelMarkerSupport.publishToProblemsViewWithoutClearing(project, projectToErrors.get(project), monitor);
+            }
+        }
+    }
+    
+    // maps the specified errors to the project instances they belong to, and returns that mapping
+    static Multimap<IProject, BazelBuildError> assignErrorsToOwningProject(List<BazelBuildError> errors,
+            Map<BazelLabel, IProject> labelToProject, IProject rootProject) {
+        Multimap<IProject, BazelBuildError> projectToErrors = HashMultimap.create();
+        List<BazelBuildError> remainingErrors = new LinkedList<>(errors);
+        for (BazelBuildError error : errors) {
+            BazelLabel owningLabel = error.getOwningLabel(labelToProject.keySet());
+            if (owningLabel != null) {
+                IProject project = labelToProject.get(owningLabel);
+                projectToErrors.put(project, error.toErrorWithRelativizedResourcePath(owningLabel));
+                remainingErrors.remove(error);
+            }
```

Comment:
we should have an else here, and attach unclaimed errors to the Bazel Workspace project